### PR TITLE
cleanup: Rename CustomSearchDomain to CustomSearchDomainConfig

### PR DESCRIPTION
**Pull request by Devinwong**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Rename CustomSearchDomain to CustomSearchDomainConfig to ensure consistency.

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Review by lilypan26 — APPROVED** [review]

File: pkg/nbcontracthelper/helper.go [hunk 1]
```diff
@@ -1,9 +1,43 @@
...
+	initializeIfNil(&v.HttpProxyConfig)
+	initializeIfNil(&v.CustomCloudConfig)
+	initializeIfNil(&v.CustomSearchDomainConfig)
+
+	return v
+}
+
+// Creates a new instance of nbcontractv1.Configuration and ensures that
+// all fields are non-nil unless it's set as proto3 explicit presence (with label optional).
+func NewNBContractConfiguration(v *nbcontractv1.Configuration) *nbcontractv1.Configuration {
```
Comment (lilypan26) [inline pkg/nbcontracthelper/helper.go:41]: As per my comment on the other PR, can we make this not take the config as an input and instead initialize an instance of the contract here?

Comment (Devinwong) [reply pkg/nbcontracthelper/helper.go]: Totally agreed. Updated.

File: pkg/nbcontracthelper/helper.go [hunk 2]
```diff
@@ -1,9 +1,74 @@
...
+	initializeIfNil(&nBCB.nBContractConfiguration.CustomLinuxOsConfig)
+	initializeIfNil(&nBCB.nBContractConfiguration.HttpProxyConfig)
+	initializeIfNil(&nBCB.nBContractConfiguration.CustomCloudConfig)
+	initializeIfNil(&nBCB.nBContractConfiguration.CustomSearchDomainConfig)
+}
+
+// Creates a new instance of NBContractConfig and ensures all objects are non-nil
+func NewNBContractBuilder() *NBContractBuilder {
+	nBCB := &NBContractBuilder{
+		nBContractConfiguration: &nbcontractv1.Configuration{
```
Comment (lilypan26) [inline pkg/nbcontracthelper/helper.go:41]: can't we just call `ensureConfigsNonNil` here?

Comment (Devinwong) [reply pkg/nbcontracthelper/helper.go]: updated

**Pull request merged**